### PR TITLE
Add requirements files

### PR DESCRIPTION
**Pull request by dp-rice**
I made one file for real requirements and "requirement-dev.txt" for mypy, black, etc. I also updated the presubmit hook to use the file. The second commit is a typo fix and should be squashed.

### CODE REVIEW

**Comment by dp-rice** [comment]
Moved line length to a pyproject.toml config file so that it's the same when I run black locally to fix my file and when it runs to check.

File: .github/workflows/presubmit.yml [hunk 1]
```diff
@@ -8,5 +8,5 @@ jobs:
         with:
             python-version: '3.11'
       - uses: actions/checkout@v3
-      - run: python3 -m pip install mypy black
+      - run: python3 -m pip install -r requirements-dev.txt
```
Comment (jeffkaufman) [inline .github/workflows/presubmit.yml:11]: You'll need `requirements.txt` here too once you add any tests that depend on `pystan`, though I guess it's fine to wait until then to sort this out.

**Review by jeffkaufman — APPROVED** [review]

File: .github/workflows/presubmit.yml (hunk 1 above)
Comment (dp-rice) [reply .github/workflows/presubmit.yml:11]: I decided to add requirements.txt to requirements-dev.txt. It's unlikely we'll ever want the dev requirements alone.

**Comment by dp-rice** [comment]
I don't mind rolling back `isort` if you don't care for it, but mypy, black, isort are my usual trio for python projects.

**Review by jeffkaufman — APPROVED** [review]

**Pull request merged**